bin/hledger-prices.hs

```diff
@@ -30,27 +32,59 @@ Print all prices from the journal.
 showPrice :: MarketPrice -> String
 showPrice mp = unwords ["P", show $ mpdate mp, T.unpack . quoteCommoditySymbolIfNeeded $ mpcommodity mp, showAmountWithZeroCommodity $ mpamount mp]
 
+divideAmount' :: Amount -> Quantity -> Amount
+divideAmount' a d = a' where
+    a' = (a `divideAmount` d) { astyle = style' }
+    style' = (astyle a) { asprecision = precision' }
+    extPrecision = (1+) . floor . logBase 10 $ realToFrac d
+    precision' = extPrecision + asprecision (astyle a)
+
```

Comment:
Is this the improve precision part ? What problem were you seeing ? Is this something that should be in the standard amount routines ?